protocol/opcm-test-integration.md

```diff
@@ -0,0 +1,130 @@
+# Direction for OPCM test integration
+
+## Purpose
+
+In order to integrate the OPCM into the forge test setup (ie. `Deploy.s.sol`), we must first
+simplify the deploy scripts while keeping the desired properties and removing what we don't need.
+
+## Goals
+
+As far as we know, these scripts are not currently being used by chain operators for deployment, and
+we intend to do that with OPCM in the future anyways. Therefore, what we need to maintain about the
+existing flow is:
+- Artifact generation:
+  - Contracts should have the same names in the resulting deploy artifacts file at
+    `/deployments/{chainId}-deploy.json`
+- EVM state:
+  - The test values configured by each test suite must be preserved (otherwise the diff to update
+    each test will explode).
+
+Nice to haves:
+- logs emitted during deployment:
+  - This is not essential, but comparing before and after logs is a quick and dirty way to ensure
+    that the config is roughly the same.
+- Deploying contracts to the same address as before.
+  - This might be achievable based on the use of
+    [`create2`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/deploy/Deploy.s.sol#L1703)
+    in the existing scripts. If so it would enable us to use statedumps to diff the before and after
+    deployments.
```

Comment:
I think this one is unlikely, because get deployed through OPCM and not from the test contract. However it is possible we can at least achieve this from the superchain and implementation deploys, so create2 with any salt can be used

Reply:
We could potentially make a small change to set the existing deployer into the same address as the OPCM will eventually go. But I don't think that is worthwhile unless we get really concerned about the ability to compare the before and after deployment.

Reply:
Ah true, that's a good idea. Alternatively we can etch OPCM into the existing deployer address. Foundry now has a `copyStorage` cheat that can ensure all storage gets migrated also: https://github.com/foundry-rs/forge-std/blob/5a802d7c10abb4bbfb3e7214c75052ef9e6a06f8/src/Vm.sol#L1552-L1553

Reply:
Nice. Will see about that as I get into it.